sfge/src/main/java/com/salesforce/rules/AbstractRuleRunner.java

```diff
@@ -86,19 +86,30 @@ private Result runPathBasedRules(
         if (rules.isEmpty()) {
             return new Result();
         }
-        List<MethodVertex> pathEntryPoints = PathEntryPointUtil.getPathEntryPoints(g, targets);
-        if (pathEntryPoints.isEmpty()) {
+        List<ApexPathSource> sources = ApexPathSource.getApexPathSources(g, rules, targets);
+        if (sources.isEmpty()) {
             LOGGER.info("No path-based entry points found");
             return new Result();
         }
 
         // Let listener know that we have finished identifying entry points in target
-        ProgressListenerProvider.get().pathEntryPointsIdentified(pathEntryPoints.size());
+        ProgressListenerProvider.get().pathEntryPointsIdentified(sources.size());
 
         // For each entry point, generate a submission object.
         List<ThreadableRuleExecutor.ThreadableRuleSubmission> submissions = new ArrayList<>();
-        for (MethodVertex pathEntryPoint : pathEntryPoints) {
-            submissions.add(getRuleRunnerSubmission(g, pathEntryPoint, rules));
+        for (ApexPathSource source : sources) {
+            // If the source was explicitly requested, run all rules against it.
+            // Otherwise, just run the rules that express interest in it.
+            List<AbstractPathBasedRule> interestedRules =
+                    source.isForceTargeted()
```

Comment:
As mentioned elsewhere, a method being specifically targeted means all rules consider it of interest. Otherwise, rules just care about sources they're interested in.